src/aus.h

```diff
@@ -245,9 +246,64 @@ namespace librealsense
             }
         }
 
+        void on_process_frame(const rs2::frame & f, std::string ppf_name)
+        {
+            std::string device_name = ((frame_interface*)f.get())->get_sensor()->get_device().get_info(RS2_CAMERA_INFO_NAME);
+            if ( _ppf_used_per_device.find(device_name) == _ppf_used_per_device.end())
+            {
+                std::set<std::string> new_set = {ppf_name};
+                _ppf_used_per_device.insert(std::make_pair(device_name, new_set));
+            }
+            else
+            {
+                _ppf_used_per_device[device_name].insert(ppf_name);
+            }
+        }
+
+        void on_start_stream(int uniqe_id)
+        {
+            std::time_t current_time = std::chrono::system_clock::to_time_t(
+                std::chrono::system_clock::now());
+            std::string str_current_time = std::ctime(&current_time);
+            _unique_profile_id_per_stream[uniqe_id] = str_current_time.substr(11, 8);
+        }
+
+        void on_stop_stream(int uniqe_id, std::string device_name, std::string sensor_name)
+        {
+            std::time_t _stop_time = std::chrono::system_clock::to_time_t(
+                std::chrono::system_clock::now());
+            std::string str_current_time = std::ctime(&_stop_time);
+            if (_sensors_per_device.find(sensor_name) == _sensors_per_device.end())
+            {
+                std::unordered_map < std::string, std::vector<std::pair<std::string, std::string>>> new_map;
+                std::vector<std::pair<std::string, std::string>> new_list;
+                new_list.push_back(std::make_pair(_unique_profile_id_per_stream[uniqe_id], str_current_time.substr(11, 8)));
+                new_map.insert(std::make_pair(device_name, new_list));
+                _sensors_per_device.insert(std::make_pair(sensor_name, new_map));
+            }
+            else
+            {
+                if (_sensors_per_device[sensor_name].find(device_name) == _sensors_per_device[sensor_name].end())
+                {
+                    std::vector< std::pair<std::string, std::string>> new_list;
+                    new_list.push_back(std::make_pair(_unique_profile_id_per_stream[uniqe_id], str_current_time.substr(11, 8)));
+                    _sensors_per_device[sensor_name].insert(std::make_pair(device_name, new_list));
+                }
+                else
+                {
+                    _sensors_per_device[sensor_name][device_name].push_back(std::make_pair(_unique_profile_id_per_stream[uniqe_id], str_current_time.substr(11, 8)));
+    
+                }     
+            }
+            _unique_profile_id_per_stream.erase(uniqe_id);
+        }
+
         std::vector<std::uint8_t>  get_data()
         {
             json j;
+            _jason_creation_time = std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
```

Comment:
Typo - should **json**
please search for all occurrences of this parameter and fix